java/vector/src/main/java/org/apache/arrow/vector/IntervalYearVector.java

```diff
@@ -125,10 +126,8 @@ public Period getObject(int index) {
       return null;
     } else {
       final int interval = valueBuffer.getInt(index * TYPE_WIDTH);
-      final int years = (interval / org.apache.arrow.vector.util.DateUtility.yearsToMonths);
-      final int months = (interval % org.apache.arrow.vector.util.DateUtility.yearsToMonths);
-      final Period p = new Period();
-      return p.plusYears(years).plusMonths(months);
+      // TODO: verify interval is in months
+      return Period.ofMonths(interval);
```

Comment:
This change here is a change in API. It will break all applications using this getObject API.